deltachat-ios/Controller/WebxdcViewController.swift

```diff
@@ -227,10 +211,16 @@ class WebxdcViewController: WebViewViewController {
             }
         }
     }
-    
-    private func updateWebxdc(statusId: Int) {
-        let statusUpdates = self.dcContext.getWebxdcStatusUpdates(msgId: messageId, statusUpdateId: statusId)
-        logger.debug("status updates: \(statusUpdates)")
+
+    var lastSerial: Int = 0
+    private func updateWebxdc() {
```

Comment:
> why not do this in js like on the other platforms?

see commit message, however, also in more detail here:

handling `lastSerial` inside js would require to keep the bit complicated `async_calls` array; moreover,  also  `getStatusUpdates` would become more complex as it would require a second parameter (`invocationId` and `lastSerial` then). doable, but all that sounds a bit too complicated without much benefits.

handling `lastSerial` inside swift removes the  `async_calls`, it removes `invocationId` array and also simplifies the call to swift (getStatusUpdates is no longer needed)

so, i think, the swift approach makes the javascript part much smaller, and removes several lines of code and complexity in general. 

> also is the swift lastSerial cleared correctly?

lastSerial is set when `setUpdateListener` is set - after that, it is set to max_serial whenever updates come in. when should it be cleared?

Reply:
when switching apps, I don't know how long those swift vars live or whether they are reused... so I thought it would be better to ask not that we miss sth.